app/index.tsx

```diff
@@ -1,67 +1,91 @@
 import { StatusBar } from 'expo-status-bar';
-import { Keyboard, ScrollView, StyleSheet, View } from 'react-native';
+import { Keyboard, StyleSheet, View, Text, ScrollView } from 'react-native';
 import NavButton from './components/NavButton';
 import PageHeader from './components/Header';
 import LabeledTextInput from './components/LabeledTextInput';
-import { getMatchData, updateMatchNumber, updateName, updateTeamNumber, updateDriverStation } from './api/data';
-import { DRIVER_STATION } from './api/data_types';
-import React, { useEffect, useState } from 'react';
+import { getMatchData, updateMatchNumber, updateName, updateTeamNumber, updateDriverStation, addUnsyncedData } from './api/data';
+import { DRIVER_STATION, MatchData } from './api/data_types';
+import { useEffect, useState } from 'react';
 import RadioButton from './components/RadioButton';
+import { onValue, ref, set } from 'firebase/database';
+import { db } from '../firebaseConfig';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import * as FileSystem from "expo-file-system";
 
 export default function App() {
     const [ nameFilled, setNameFilled ] = useState(false);
     const [ teamNumberFilled, setTeamNumberFilled ] = useState(false);
     const [ matchFilled, setMatchFilled ] = useState(false);
+    const [ eventCode, setEventCode ] = useState("");
 
     useEffect(() => {
-        const initializeRequirements = () => {
-            getMatchData().then((data) => {
-                setNameFilled(data["scouterName"] !== "");
-                setTeamNumberFilled(data["teamNumber"] !== 0);
-                setMatchFilled(data["matchNumber"] !== 0);
-            })
-        }
+        getMatchData().then((data) => {
+            setNameFilled(data["scouterName"] !== "");
+            setTeamNumberFilled(data["teamNumber"] !== 0);
+            setMatchFilled(data["matchNumber"] !== 0);
+        })
 
-        initializeRequirements();
-    })
+        onValue(ref(db, ".info/connected"), (snap) => {
```

Comment:
If you want to split this up, you can sync event code to Async Storage, and use Event Code from Async Storage separately (will reduce nesting). This is fine too though.

Reply:
i don't think this would reduce the nesting, since you'd need to await the event code from AsyncStorage anyways.